### ISSUE TITLE
[All fields and inputs] Helper text update

### ISSUE THREAD

**Issue by jeanservaas** [carbon-design-system/carbon#2329]
#### Jan Child, the Design Lead for Watson assistant made several really good points about the way we're treating helper text followed by these suggestions...

This issue relates to all of our input fields (i.e. text input, text area, number input, select and dropdown components); it also affects the $helper-text type token (which would eventually need to be deprecated).

> The Watson Assistant team would like to the helper text to be below the input field rather than above it (directly under the label); when the label is underneath, it's only there if you need it vs. when it's above, it's given the same weight as the label and is obtrusive for the user.

- ~~Move helper text below the input / dropdown field~~ Edit based on thread: doesn't work coz dropdown opens and covers helper text.

> They also want to eliminate the use of italics; if the helper text is below the input field, there's no need for the italic to differentiate it from the label; also, the use of italics in the situation was not appropriate in the first place. Italics should be reserved primarily for single-word emphasis, titles / names and foreign / technical words and terms.

- ~~Use $label-01 token for helper text (eliminates the italic); eventually deprecate $helper-text-01; in the meantime, update $helper-text-01 to call regular Plex Sans instead of italic ~~ Edit based on thread: Keeping italic because it is currently the only visual differentiation between label text and helper text. 

> Speaking for the design team; I agree with these proposed changes; due to time constraints, we didn't spend enough time iterating on the helper text designs. Our primary concern was that it not interfere with the error text below the input field. Jan suggested that best practice here would be to remove the helper text when the warning appears.

- ~~In the instance of an error state, remove helper text in favor of error message~~ Edit based on thread: no longer relevant if we don't move text down. 

## Designs:

<br>

Current state of helper text in context:

![image.png](https://images.zenhubusercontent.com/5b96e742f28d730b01a470c0/87e59d91-bd98-475b-a9c2-7c79339fa374)

<br>

Proposed change:

![image.png](https://images.zenhubusercontent.com/5b96e742f28d730b01a470c0/7291f2c0-09fe-418d-b18c-bf8aa64003b4)

**Comment by elizabethsjudd** [comment]
From an a11y side of things moving the helper text would mean that the `aria-describedby` attribute would need to be used on the input that is connected to the `id` of the helper text so that screen reader users have access to the helper text when they are on the input and not after they leave the input.

Or even easier, using Flexbox's order property.  Here's a screen shot from one of my team member's playing around with the component.

<img width="945" alt="Screen Shot 2019-04-19 at 4 09 48 PM" src="https://user-images.githubusercontent.com/1275779/56442646-920b4b00-62bf-11e9-8bf9-2d180f7545cd.png">

**Comment by jeanservaas** [comment]
Thanks for this input @elizabethsjudd! Just to clarify, do you see this as an impediment to making this change? Or is this just an added bullet point for the devs to tackle?

**Comment by elizabethsjudd** [comment]
@jeanservaas simply something for the devs to tackle while implementing the change.  It shouldn't block the update.

**Comment by jeanservaas** [comment]
THIS IS NOT STALE

**Comment by janchild** [comment]
Note to @shixiedesign and @jeanservaas and @lovemecomputer ... I've been looking into this further and we need to hold before making this change as it's going to be a problem for accessibility. My apologies for the churn. 

I'm looking at this further tomorrow with Tom Roach and we'll come back with refinements.

**Comment by elizabethsjudd** [comment]
@janchild it is possible to do this and have it still be accessible.  See my comment above for two different options.

**Comment by janchild** [comment]
Hi @elizabethsjudd, thank you for that information! Tom Roach and I have been working through this, and after much discussion would prefer to keep the helper text _above_ the field. There were some other issues that we uncovered when we stepped through other UI elements:

-  When the helper text is needed for lists, it makes sense to include it under the label and above the list. If positioned beneath a longer list, you wouldn't see it until you read to the bottom. 

- Also for dropdowns it was an issue because the helper text is obscured as soon as the dropdown is opened and the user may need to read it to make a selection. 

- We also felt that a consistent approach for all elements would make it easier for all to implement, rather than having to explain why a field has a different treatment. 

The only change then we would be looking for is to remove the italics.

**Comment by elizabethsjudd** [comment]
@janchild thanks for the transparency on your thought process. I completely agree with all of those statements!

**Comment by janchild** [comment]
Fantastic, @elizabethsjudd ! Thank you.

**Comment by shixiedesign** [comment]
Removing italics will cause label and helper text to look exactly the same, so that's not an improvement. A few other proposals: 

1. Gray 50 helper text, 3:1 accessible against White& Gray10, obv still use `aria-describedby` 

![image](https://user-images.githubusercontent.com/15144993/57869118-f7246480-77c9-11e9-9fb8-398969e16ecc.png)

2. Go with icon tooltip for all

![image](https://user-images.githubusercontent.com/15144993/57869700-369f8080-77cb-11e9-8b0f-f98694d3e0b2.png)

**Comment by janchild** [comment]
Good point, @shixiedesign. I really like option #1.

We will be using icon tooltips but more sparingly, so Gray 50 helper text always visible is a great solution. Thank you!

**Comment by shixiedesign** [comment]
Unable to proceed with option 1 because helper text has to be 4.5:1 accessible against background. So far, it doesn't seem like we have a better alternative than italic text, and removing the italic styling would mean a regression. 

I will wait on this issue for a bit longer for more solution / proposals. If we don't have a good alternative, issue will be closed.

**Comment by shixiedesign** [comment]
Okay one last option from me. If we use the **new palette** I proposed here https://github.com/carbon-design-system/carbon/issues/2729#issuecomment-495470891 , we have accessible colors for 3 tiers of text hierarchy: on light Gray 100, Gray 80, Gray 60; on dark Gray 10, Gray 30, Gray 50. 

We can use the 3rd tier / lightest color for helper text, 2nd tier for labels, and 1st tier for body copy. 

![image](https://user-images.githubusercontent.com/15144993/58304430-d0fa5800-7db9-11e9-8f49-732a48ba2617.png)

**Comment by shixiedesign** [comment]
Palette work in progress.

**Comment by shixiedesign** [comment]
Blocked by https://github.com/carbon-design-system/carbon/issues/3226

**Comment by shixiedesign** [comment]
not stale

**Comment by jeanservaas** [comment]
Keep open to wait for palette work

**Comment by lnajjar** [comment]
It looks like you're going this way anyway, but, if not, here is a rationale. To match users' cognitive flow and prevent user input errors, it would be helpful to retain your original design: Label, helper text, input field, error message. That way, users read the label, then the helper text, then make a valid entry. As you do now, displaying the helper text in italics might be a nice way to differentiate the helper text from the label.

**Comment by shixiedesign** [comment]
Closing this issue to start a new one to track implementation of color palette solution. See https://github.com/carbon-design-system/carbon/issues/3961
